extension/goplugin/types_test.go

```diff
@@ -0,0 +1,134 @@
+// Copyright (C) 2017 NTT Innovation Institute, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goplugin_test
+
+import (
+	"encoding/json"
+
+	"github.com/cloudwan/gohan/extension/goext"
+	"github.com/cloudwan/gohan/extension/goplugin"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Types tests", func() {
+	var (
+		env *goplugin.Environment
+	)
+
+	BeforeEach(func() {
+		env = goplugin.NewEnvironment("test", nil, nil)
+	})
+
+	AfterEach(func() {
+		env.Stop()
+	})
+
+	Describe("JSON Marshalling", func() {
```

Comment:
Please add some tests for the specific equality operator to ensure no future changes break its definition.